internal/client/managementclusterregistration/management_cluster_registration_resource.go

```diff
@@ -38,43 +42,48 @@ func New(transport *transport.Client) ClientService {
 
-	ManagementClusterResourceReregisterService(params *registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterRequest) (*registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse, error)
+	ManagementClusterResourceServiceDelete(request *registration.VmwareTanzuManageV1alpha1ManagementclusterFullName, force string) error
 
-	ManagementClusterResourceServiceDelete(params *registration.ManagementClusterResourceServiceFullName) error
+	ManagementClusterResourceServiceGet(request *registration.VmwareTanzuManageV1alpha1ManagementclusterFullName) (*registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse, error)
 
-	ManagementClusterResourceServiceGet(params *registration.ManagementClusterResourceServiceFullName) (*registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse, error)
+	ManagementClusterResourceServiceUpdate(reguest *registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterRequest) (*registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse, error)
 
-	ManagementClusterResourceServiceUpdate(request *registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterRequest) (*registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse, error)
+	ManagementClusterManifestHelperGetManifest(request *registration.VmwareTanzuManageV1alpha1ManagementclusterFullName) (*registration.VmwareTanzuManageV1alpha1ManagementclusterManagementClusterGetManifestResponse, error)
 }
 
 /*
-ManagementClusterResourceServiceCreate creates a management cluster
+ManagementClusterResourceServiceCreate creates a management cluster registration
 */
 func (c *Client) ManagementClusterResourceServiceCreate(request *registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterRequest) (*registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse, error) {
 	namespaceResponse := &registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse{}
 	err := c.Create(apiVersionAndGroup, request, namespaceResponse)
 
-	//TODO if err is returned with value 6 then execute method ManagementClusterResourceReregisterService - this will be implemented in the next PR
-
 	return namespaceResponse, err
 }
 
+/*
+ManagementClusterResourceReregisterService reregisters management cluster registration
+*/
 func (c *Client) ManagementClusterResourceReregisterService(request *registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterRequest) (*registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse, error) {
-	namespaceResponse := &registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse{}
+	reregisterResponse := &registration.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse{}
```

Comment:
nit: reregisterResponse? Could you please elaborate on the naming used?

Reply:
I probably got  attached to the name of this method name instead of actual model and usage. Renamed it so it won't have extra noise.